src/net/sourceforge/kolmafia/StaticEntity.java

```diff
@@ -643,26 +639,12 @@ public static final void generateHeapDump() {
   }
 
   public static final String[] getPastUserList() {
-    ArrayList<String> pastUserList = new ArrayList<String>();
-
-    String user;
-    File[] files = DataUtilities.listFiles(KoLConstants.SETTINGS_LOCATION);
-
-    for (int i = 0; i < files.length; ++i) {
-      user = files[i].getName();
-      if (user.startsWith("GLOBAL") || !user.endsWith("_prefs.txt")) {
-        continue;
-      }
-
-      user = user.substring(0, user.length() - 10);
-      if (!user.equals("GLOBAL") && !pastUserList.contains(user)) {
-        pastUserList.add(user);
-      }
-    }
-
-    String[] pastUsers = new String[pastUserList.size()];
-    pastUserList.toArray(pastUsers);
-    return pastUsers;
+    return Arrays.stream(DataUtilities.listFiles(KoLConstants.SETTINGS_LOCATION))
+        .map(f -> f.getName())
+        .filter(u -> !u.startsWith("GLOBAL") && u.endsWith("_prefs.txt"))
+        .map(u -> u.substring(0, u.length() - 10))
```

Comment:
You seem to have lost the "user != 'GLOBAL'" check.

Might be intended because surely that's covered by "starts with GLOBAL", but worth checking.

Reply:
Yeah I did it on purpose, it was redundant